pkg/gen/awaitComments.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gen
+
+import (
+	"github.com/pulumi/pulumi-kubernetes/pkg/kinds"
+)
+
+const DeploymentAwaitComment = `Pulumi uses "await logic" to determine if a Deployment is ready.
+The following conditions are considered by this logic:
+1. The Deployment has begun to be updated by the Deployment controller. If the current
+   generation of the Deployment is > 1, then this means that the current generation must
+   be different from the generation reported by the last outputs.
+2. There exists a ReplicaSet whose revision is equal to the current revision of the
+   Deployment.
+3. The Deployment's '.status.conditions' has a status of type 'Available' whose 'status'
+   member is set to 'True'.
+4. If the Deployment has generation > 1, then '.status.conditions' has a status of type
```

Comment:
It feels like this 4th point should be nested in point 3 given that its about `.status.conditions`

Reply:
It's checking separate status conditions, so I'll leave it as is.